pkg/build/generator/generator.go

```diff
@@ -170,6 +170,10 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 		return nil, err
 	}
 
+	if bc.DeletionTimestamp != nil {
+		return nil, fmt.Errorf("Cannot generate Build from BuildConfig %s/%s because it is being deleted", bc.Namespace, bc.Name)
```

Comment:
this needs to be a status error since it goes back out over the rest api, maybe InternalError: https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/errors/errors.go#L236